addons/sale/models/sale.py

```diff
@@ -501,6 +529,11 @@ def _create_invoices(self, grouped=False, final=False):
         invoices_name = {}
 
         for order in self:
+            if order.has_to_be_down_paid():
```

Comment:
I am not sure it should be done here. Shouldn't the caller check that it is effectively in a case where invoices can be created before calling _create_invoices instead of simply skipping it without notifying anybody ? Purpose of this method is to create invoice based on SO. I think caller should ensure it is doable. This method should have standard I/O .

Reply:
cc @jem-odoo the man who knows everything about invoice / so confirmation (or not, maybe)

Reply:
Yes, I agree with tde ! Otherwise, invoice for given SO will not be created and the user is not warned. I think we can say that all record in `self` should be invoicable, as prerequisite.

Reply:
Two cases : 
1) Draft/Sent : if the user pays for the SO, it is to confirm it and as the specs says, no invoice should be generated for the partial payment of confirmation
2) Confirmed : has_to_be_paid/down_paid == False and invoices are always generated
Invoices are never generated for unconfirmed quotations (at least not with the current UI).

Therefore, I only avoid the case where "automatic invoice" is checked and a payment is done for the SO so that it doesn't generate an invoice (a warning is added in the settings on this subject).

@jem-odoo Where could I catch the invoice creation otherwise when a partial confirmation payment is done (as fgi wants to avoid having invoices in this case).

Reply:
Moved to payment.py in `_invoice_sale_orders` to avoid modifying the general `_create_invoices` in sale.py.